components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts

```diff
@@ -51,19 +61,34 @@ export class ProjectsFilterDropdownComponent implements OnChanges {
     }
   }
 
-  handleEscape() {
-    this.dropdownActive = false;
+  handleEscape(): void {
     this.resetOptions();
+    this.filterValue = '';
+    this.dropdownActive = false;
+    this.onOptionChange.emit(this.editableOptions);
   }
 
-  handleOptionChange(event, index) {
-    this.editableOptions[index].checked = event.detail;
+  handleOptionChange(event, label) {
+    this.editableOptions.find(option => {
+      if (option.label === label) {
+        option.checked = event.detail;
```

Comment:
what's `detail`? is that a generic event thing?

Reply:
@bcmdarroch - in this particular case, `detail` returns either a `true` or `false` depending on the state of the checkbox being selected.